shakedown/dcos/package.py

```diff
@@ -87,51 +87,55 @@ def install_package(
 
     package_manager = _get_package_manager()
     pkg = package_manager.get_package_version(package_name, package_version)
+
     if service_name is None:
-        labels = pkg.marathon_json(options).get('labels')
-        if 'DCOS_SERVICE_NAME' in labels:
-            service_name = labels['DCOS_SERVICE_NAME']
+        try:
+            labels = pkg.marathon_json(options).get('labels')
+            if 'DCOS_SERVICE_NAME' in labels:
+                service_name = labels['DCOS_SERVICE_NAME']
+        except errors.DCOSException as e:
+            pass
```

Comment:
This leaves the possibility open that `service_name` is still `None`. Should this print a shakedown error and exit instead of continuing onward?

Reply:
Yes, it's okay for `service_name` to be `None`; in cases like `dcos-enterprise-cli`, there is no "app" or "service", so this will always be `None`.

`dcos.package_manager.marathon_json` will throw an exception for packages that do not contain Marathon definitions (like `dcos-enterprise-cli`), which is what this try/except block is catching.